src/index.tsx

```diff
@@ -157,9 +148,42 @@ var saveToWebmarkFolder = (url: string | undefined, title: string | undefined):
     );
 }
 
-var getRandomUrlFromFolder = (): string => {
-    //TODO: implement getRandomUrlFromFolder
-    return 'https://example.com/random';
+let getRandomUrlFromFolder = (): void => {
+    chrome.storage.sync.get(
+        ['webmarkFolderId'],
+        (result?) =>{
+            console.log('webmarkFolderId found.');
+            let webmarkFolderId: string = result!['webmarkFolderId'];
+            chrome.bookmarks.getSubTree(
+                webmarkFolderId,
+                (bookmarkTreeNodes) =>{
+                    let urlList : Array<string> = [];
+                    for (let node of bookmarkTreeNodes) {
+                        if(node.children != undefined){
+                            for(let child of node.children){
+                                if(child.url!=undefined){
+                                    urlList.push(child.url);
+                                }
+                            }
+                        }
+                    }
+                    let randomUrl : string= urlList[Math.floor(Math.random() * urlList.length)];
+                    chrome.storage.sync.get(
+                        ['loadHere'],
+                        function (result) {
+                            if (Object.keys(result).length != 0 && result!['loadHere']) {
+                                loadInCurrentTab(randomUrl);
+                            }
+                            else {
+                                loadInNewTab(randomUrl);
+                            }
+                        }
+                    );
+                }
+            );
+        }
+    );
+    
```

Comment:
Intended?
Besides, many lines changed once I formatted the code. Are you sure you set `"editor.formatOnSave": true`?

Reply:
I am using that option. What should it look like? I don't quite understand.

Reply:
We might be using different formatter too.

Reply:
Turns out the culprit was:
```
"[javascript]": {
        "editor.defaultFormatter": "vscode.typescript-language-features"
    },
```

Reply:
was it?